### PR TITLE
Fix/schedule updates

### PR DESCRIPTION
**Pull request by mazenchami**
# Description

[Trello Card](153-app-is-busted-now-on-first-load) — #119 & #121

After WebFlow was updated with the new schedule related fields, the app broke. This PR fixes those issues and cleans up the return in our `webflow-helpers.ts` file to display the `ScheduleCard`'s correctly.

# Graphics

| iOS            | Android            |
| -------------- | ------------------ |
|  |  |

# Checklist:

- [x] I have done a thorough self-review of my code
- [ ] I have tested with a screen reader and font-scaling turned on and added necessary accessibility features
- [x] I have run tests and linter

### CODE REVIEW

File: app/services/api/webflow-helpers.ts [hunk 1]
```diff
@@ -118,17 +118,30 @@ export const cleanedWorkshops = (
...
           schedule["day-time"],
-          schedule["end-time"] ? "h:mm" : "h:mm aaa",
+          schedule["end-time"] && !isTheSameTime ? "h:mm" : "h:mm aaa",
         ),
         startTime: schedule["day-time"],
-        formattedEndTime: schedule["end-time"] && formatDate(schedule["end-time"], "h:mm aaa"),
-        eventTitle: schedule["recurring-event"]?.name,
-        heading: "",
+        formattedEndTime:
+          schedule["end-time"] && !isTheSameTime && formatDate(schedule["end-time"], "h:mm aaa"),
```
Comment (morganick) [inline app/services/api/webflow-helpers.ts:142]: This makes me wonder if we shouldn't have a `formattedTime` attribute that automatically handles the same time and formats it appropriately.

File: app/services/api/webflow-helpers.ts [hunk 2]
```diff
@@ -191,7 +209,12 @@ export const convertScheduleToScheduleCard = (
   day: string,
 ): ScheduleCardProps[] => {
   const daySchedule: ScheduledEvent[] = groupBy("day")(scheduleData ?? [])?.[day] ?? []
-  return daySchedule.sort(sortByTime).map(convertScheduleToCardProps)
+  const sortedDaySchedule = daySchedule
+    .sort(sortByTime)
+    .map(convertScheduleToCardProps)
+    .filter(Boolean)
+  console.tron.log({ day, daySchedule: daySchedule.sort(sortByTime), sortedDaySchedule })
```
Comment (joshuayoes) [inline app/services/api/webflow-helpers.ts:216]: We will probably want to remove this before merging, or hide behind __DEV__ flag

File: app/services/api/webflow-helpers.ts [hunk 3]
```diff
@@ -173,12 +187,16 @@ const convertScheduleToCardProps = (schedule: ScheduledEvent): ScheduleCardProps
       const workshop = schedule.workshop
       return {
         variant: "workshop",
-        formattedStartTime: formatDate(schedule["day-time"], "h:mm aaa"),
+        formattedStartTime: formatDate(
+          schedule["day-time"],
+          schedule["end-time"] ? "h:mm" : "h:mm aaa",
```
Comment (morganick) [inline app/services/api/webflow-helpers.ts:192]: Are `Recurring` types the only ones we care about augmenting the date format? It looks like we're making the assumption that they will never be the same.

File: app/services/api/webflow-helpers.ts [hunk 4]
```diff
@@ -191,7 +209,12 @@ export const convertScheduleToScheduleCard = (
   day: string,
 ): ScheduleCardProps[] => {
   const daySchedule: ScheduledEvent[] = groupBy("day")(scheduleData ?? [])?.[day] ?? []
-  return daySchedule.sort(sortByTime).map(convertScheduleToCardProps)
+  const sortedDaySchedule = daySchedule
```
Comment (morganick) [inline app/services/api/webflow-helpers.ts:212]: Was this just for debugging?

**Review by joshuayoes — APPROVED** [review]
This builds on my branch! I've merged it into my other Zod PR here: https://github.com/infinitered/ChainReactApp2023/pull/125

File: app/services/api/webflow-helpers.ts (hunk 2 above)
Comment (joshuayoes) [reply app/services/api/webflow-helpers.ts:216]: I did remove this console.tron call in my other branch: https://github.com/infinitered/ChainReactApp2023/pull/125

File: app/services/api/webflow-helpers.ts (hunk 1 above)
Comment (mazenchami) [reply app/services/api/webflow-helpers.ts:142]: totally open if you'd like to branch off of this and handle that?

File: app/services/api/webflow-helpers.ts (hunk 4 above)
Comment (mazenchami) [reply app/services/api/webflow-helpers.ts:212]: correct, removing in next push

File: app/services/api/webflow-helpers.ts (hunk 2 above)
Comment (mazenchami) [reply app/services/api/webflow-helpers.ts:216]: its was left over from debugging, removing in next push

**Review by morganick — APPROVED** [review]
I'll look at the refactors in a separate PR.

LGTM 👍

File: app/services/api/webflow-helpers.ts (hunk 3 above)
Comment (mazenchami) [reply app/services/api/webflow-helpers.ts:192]: regarding the dates, we're deciding based on the UI. not the best but it gets the UI in place as needed

**Pull request merged**